examples/meta/src/neural_nets/feedforward_net_regression.sg

```diff
@@ -1,54 +1,45 @@
-CSVFile f_feats_train("../../data/regression_1d_sinc_features_train_with_9d_noise.dat")
-CSVFile f_feats_test("../../data/regression_1d_sinc_features_test_with_9d_noise.dat")
-CSVFile f_labels_train("../../data/regression_1d_sinc_labels_train.dat")
-CSVFile f_labels_test("../../data/regression_1d_sinc_labels_test.dat")
+File f_feats_train = csv_file("../../data/regression_1d_sinc_features_train_with_9d_noise.dat")
+File f_feats_test = csv_file("../../data/regression_1d_sinc_features_test_with_9d_noise.dat")
+File f_labels_train = csv_file("../../data/regression_1d_sinc_labels_train.dat")
+File f_labels_test = csv_file("../../data/regression_1d_sinc_labels_test.dat")
 
 Math:init_random(1)
 
 #![create_features]
 Features features_train = features(f_feats_train)
 Features features_test = features(f_feats_test)
-RegressionLabels labels_train(f_labels_train)
-RegressionLabels labels_test(f_labels_test)
+Labels labels_train = labels(f_labels_train)
+Labels labels_test = labels(f_labels_test)
 #![create_features]
 
 #![add_layers]
-int dimensions = features_train.get_int("num_features")
-NeuralLayers layers() 
-layers.input(dimensions)
-layers.rectified_linear(20)
-layers.linear(1)
-DynamicObjectArray all_layers = layers.done()
+int num_feats = features_train.get_int("num_features")
+DynamicObjectArray all_layers()
```

Comment:
btw instead of using the array type here explicitly, we should be able to use `SGObject::add`, which allows to append to an internal array.
The call should be `machine.add("layers", input);` etc. This means you would first create the network instance, and only then add the layers.
If this works, we dont need to add the array type to shogun's base classes, which I would prefer

Reply:
see the multiple kernel learning example where the same thing is done